vdj_ann/src/transcript.rs

```diff
@@ -5,29 +5,49 @@
 use crate::annotate::get_cdr3_using_ann;
 use crate::refx::RefData;
 use amino::{have_start, have_stop};
-use debruijn::{dna_string::DnaString, kmer::Kmer20, Mer, Vmer};
+use debruijn::dna_string::DnaString;
+use debruijn::kmer::Kmer20;
+use debruijn::{Mer, Vmer};
 use hyperbase::Hyper;
 use io_utils::fwriteln;
 use kmer_lookup::make_kmer_lookup_20_single;
-use std::{cmp::max, io::prelude::*};
+use std::cmp::max;
+use std::io::prelude::*;
 use vector_utils::{lower_bound1_3, unique_sort};
 
 // ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
 // TEST FOR VALID VDJ SEQUENCE
 // ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
 
+#[derive(Debug, PartialEq, Ord, PartialOrd, Eq)]
+pub enum UnproductiveContigCause {
+    NoCdr3,
+    Misordered,
+    NotFull,
+    TooLarge,
+}
```

Comment:
Can we add comments above each failure state describing exactly what criteria need to met to qualify.